src/main/java/com/exadel/discount/service/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.exadel.discount.service;
+
+import com.exadel.discount.model.token.TokenPayload;
+import com.exadel.discount.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service("userDetailsServiceImpl")
```

Comment:
why do you need this name?

Reply:
There was no need, sorry. It was deleted.